src/module.c

```diff
@@ -2394,6 +2394,24 @@ void RM_Yield(RedisModuleCtx *ctx, int flags, const char *busy_reply) {
             /* Let redis process events */
             processEventsWhileBlocked();
         } else {
+            /* If it is not the main thread, we need to process events after the
+             * main thread enter acquiring GIL state in order to protect the ae and
+             * avoid potential race conditions. */
+            if (!pthread_equal(server.main_thread_id, pthread_self())) {
+                int acquiring;
+                atomicGet(server.module_gil_acquring, acquiring);
+                if (!acquiring) {
+                    /* If the main thread has not yet entered the acquiring GIL state,
+                     * we attempt to wake it up and exit without waiting for it to
+                     * acquire the GIL. This avoids blocking the caller, allowing them to
+                     * continue with unfinished tasks before the next yield. */
+                    if (write(server.module_pipe[1],"A",1) != 1) {
+                        /* Ignore the error, this is best-effort. */
+                    }
+                    goto end;
```

Comment:
Using goto is highly discouraged.

It leads to spaghetti code and unmaintainable mess.

Please always refactor such codes. Also, if you find existing occurrences,, kindly correct them as well.

Always strive to put things in a better state.

Reply:
there is nothing wrong with `goto` if we using it properly.
We can quickly jump out of the loop, or clean up before leaving method, which is much cleaner than using intermediate variables or using `do {} while (0)`, isn't it?